pkg/report/linux.go

```diff
@@ -859,6 +859,44 @@ func (ctx *linux) isCorrupted(title string, report []byte, format oopsFormat) (b
 	return false, ""
 }
 
+func (ctx *linux) reportType(rep *Report) Type {
+	if strings.HasPrefix(rep.Title, "INFO: rcu detected stall") ||
+		strings.HasPrefix(rep.Title, "INFO: task hung") ||
+		strings.HasPrefix(rep.Title, "BUG: soft lockup") ||
+		strings.HasPrefix(rep.Title, "INFO: task can't die") {
+		return Hang
+	}
+	if strings.HasPrefix(rep.Title, "UBSAN: ") {
+		return UBSAN
+	}
+	if strings.Contains(rep.Title, "sleeping function called from invalid context") ||
+		strings.Contains(rep.Title, "scheduling while atomic") {
+		return AtomicSleep
+	}
+	if strings.Contains(rep.Title, "locking bug in") ||
```

Comment:
At this point I think report type should be part of oopsFormat.
We have massive duplication with oopsFormat tables in the form of ad-hoc imperative string matching.